backend/src/main/java/com/google/sps/servlets/SearchMatchServlet.java

```diff
@@ -0,0 +1,81 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.sps.data.Match;
+import com.google.sps.data.Participant;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+* Servlet that returns some example content.
+*/
+@WebServlet("/search-match")
+public class SearchMatchServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get participant ldap
+    UserService userService = UserServiceFactory.getUserService();
+    String email = userService.getCurrentUser().getEmail();
+    String ldap = email.split("@")[0];
+
+    // Create and sort queries by time
+    Query query = new Query("Match").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    // Convert list of entities to list of matches
+    List<Match> matches = new ArrayList<Match>();
+    for (Entity entity : results.asIterable()) {
+        long id = (long) entity.getKey().getId();
+        Participant firstParticipant = (Participant) entity.getProperty("firstParticipant");
+        Participant secondParticipant = (Participant) entity.getProperty("secondParticipant");
+        int duration = (int) entity.getProperty("duration"); 
+        long timestamp = (long) entity.getProperty("timestamp");
+        Match match = new Match(id, firstParticipant, secondParticipant, duration, timestamp);
+        matches.add(match);
+    }
+
+    Match matchDetails = new Match(System.currentTimeMillis()); // default if no match
+    // Brute force search for match
+    for (Match match : matches) {
+      if (ldap.equals(match.getFirstParticipant().getLdap()) || ldap.equals(match.getSecondParticipant().getLdap())) {
+        matchDetails = match;
+        break;
+      }
+    }
+    
+    // Send the JSON back as the response
+    Gson gson = new Gson();
+    response.setContentType("application/json");
+    response.getWriter().println(gson.toJson(matchDetails));
+  }
```

Comment:
I'm concerned about using an instance of the Match class to indicate no Match here. The second constructor in Match that takes a timestamp adds the possibility that any of the fields on Match are null / zero values. So any caller to match.getFirstParticipant() would have to check for null because it's possible and expected for some Matches to have null participant fields.

Could another object or format be used to communicate to the client that no match was found?

Reply:
I changed this part so it returns:
- matchStatus: false if it doesn't exists
- matchStatus: true, firstParticipantLdap: xx, secondParticipantLdap: xx, duration: xx if it does. 
Does this work better? Now they'd just check the value of matchStatus.